Changed the way text is rendered (Fixes: #47)

mjarkk opened the pull request:
This pr changes the way text is rendered.  
The way text wrapping works right now supper buggy.  
Beside that there are some really dumb calculations dune because some positions are based on the view offset what makes working with this a bit annoying.  

I changed most positions in the view to be based on the lines and not the actual line wrappings what makes working with then much easier and less buggy.

mjarkk commented:
@skanehira Can you re-check the wide char support?  
I think it works but i don't have any keyboard chars to test it with, only copy and past from the internet.  

@glvr182 sorry for the commit mess, lazygit pushed the merge changes to origin in-staid of my remote.

glvr182 commented:
Code looks good, @skanehira if you could test this I'll accept the code.  
As for the commits, can you reset head soft and push it all in one commit.
You can do it like this:  
```
git commit -s -m "View: Changed edit logics
 * thing you changed
 * thing you changed
"
```

mjarkk committed:
Changed the way gocui text rendering works

Most positions are now based on the lines instaid of the actual view.
This removes a lot of complicated code because we don't have to worry anymore about the view offset.

Improved line wrapping by a lot no wired cursor movement anymore.

mjarkk commented:
I'm going to wait with merging this until the switch to TCell is dune, this pr changes quite a bit core logic and if we merge it now end users might have to change there code a bit 2 times.

C0DK commented:
Cannot wait for this to be merged!

mjarkk committed:
Merge branch 'master' into fix-text-insert

mjarkk committed:
Fixed issues after merge

mjarkk committed:
Added extra information about how changes made

mjarkk commented:
@glvr182 Should we also merge this in?  
I've merged the master code and everything still seems to work fine.  
Also moved the tcell changes markdown file to `migrate-to-v1.md` and added extra information about the changes in this PR

glvr182 commented:
Sounds good! We can then test both the upgrade and this new method of rendering

mjarkk commented:
A new beta release is up with the tcell and these changes,  
[v1.0.0-beta-2](https://github.com/awesome-gocui/gocui/releases/tag/v1.0.0-beta-2)
